ncbi_genome_download/core.py

```diff
@@ -366,33 +370,33 @@ def download_entry(entry, section, domain, output, file_format, human_readable,
     else:
         formats = [file_format]
 
-    download_jobs = []
     for fmt in formats:
         try:
             if has_file_changed(full_output_dir, parsed_checksums, fmt):
-                download_jobs.append(
-                    download_file_job(entry, full_output_dir, parsed_checksums, fmt, symlink_path, table))
+                yield download_file_job(entry, full_output_dir, parsed_checksums, fmt, symlink_path, table)
             elif need_to_create_symlink(full_output_dir, parsed_checksums, fmt, symlink_path):
-                download_jobs.append(
-                    create_symlink_job(full_output_dir, parsed_checksums, fmt, symlink_path))
+                yield create_symlink_job(full_output_dir, parsed_checksums, fmt, symlink_path)
         except ValueError as err:
             logging.error(err)
 
-    return download_jobs
 # pylint: enable=too-many-arguments
 
-
-def create_dir(entry, section, domain, output):
-    """Create the output directory for the entry if needed"""
-    full_output_dir = os.path.join(output, section, domain, entry['assembly_accession'])
+def mkdir_p(path):
```

Comment:
This seems to be an unrelated change. One I'd happily consider in a separate PR, though.